frontend/src/components/Hacknight.vue

```diff
@@ -0,0 +1,65 @@
+<template>
+  <v-container fluid text-xs-center>
+    <v-row align="center" justify="center">
+      <v-col class="d-flex" cols="12" sm="6">
+        <v-select
+          v-model="selectedHacknight"
+          :items="hacknights"
+          item-text="date"
+          item-value="id"
+          label="Select Hacknight"
+          @input="onGetHacknight"
+        >
+          <template v-slot:append-outer>
+            <v-btn
+              color="blue-grey"
+              class="ma-2 white--text"
+              @click="onCreateHacknight"
+              style="top: -12px"
+              offset-y
+            >
+              <v-icon left dark>mdi-plus</v-icon>
+              New
+            </v-btn>
+          </template>
+        </v-select>
+      </v-col>
+    </v-row>
+  </v-container>
+</template>
+
+<script>
+export default {
+  data() {
+    return {
+      selectedHacknight: '',
+      hacknight: {},
+      hacknights: []
+    };
+  },
+  created() {
+    this.$store.dispatch('hacknight/getHacknights').then(res => {
+      this.hacknights = res.data.hacknights;
+    });
+  },
+  methods: {
+    onCreateHacknight() {
+      this.$store.dispatch('hacknight/createHacknight').then(res => {
+        this.hacknight = res.data.hacknight;
```

Comment:
After digging into Vuex documentation I believe we should use its store in a different way than presented here.

Probably there's no reason to have `hacknights` as a component attribute, we could use the store directly here. It would probably also allow us to use the state without explicitly handling promises here.

Reply:
We can have that conversation on next hacknight @stanislawK

Reply:
@stanislawK has that been resolved?

Reply:
I hope so - I rewrote store implementation for Hacknight